sdk/identity/Azure.Identity/src/ManagedIdentityClient.cs

```diff
@@ -102,49 +152,51 @@ public virtual MsiType GetMsiType(CancellationToken cancellationToken)
                             // if BOTH the env vars MSI_ENDPOINT and MSI_SECRET are set the MsiType is AppService
                             if (!string.IsNullOrEmpty(secretEnvVar))
                             {
-                                s_msiType = MsiType.AppService;
+                                _msiType = MsiType.AppService;
                             }
                             // if ONLY the env var MSI_ENDPOINT is set the MsiType is CloudShell
                             else
                             {
-                                s_msiType = MsiType.CloudShell;
+                                _msiType = MsiType.CloudShell;
                             }
                         }
                         // if MSI_ENDPOINT is NOT set AND the IMDS endpoint is available the MsiType is Imds
                         else if (ImdsAvailable(cancellationToken))
                         {
-                            s_endpoint = s_imdsEndpoint;
-                            s_msiType = MsiType.Imds;
+                            _endpoint = s_imdsEndpoint;
+                            _msiType = MsiType.Imds;
                         }
-                        // if MSI_ENDPOINT is NOT set and IMDS enpoint is not available ManagedIdentity is not available
+                        // if MSI_ENDPOINT is NOT set and IMDS endpoint is not available ManagedIdentity is not available
                         else
                         {
-                            s_msiType = MsiType.Unavailable;
+                            _msiType = MsiType.Unavailable;
+
+                            AzureIdentityEventSource.Singleton.ImdsEndpointUnavailable(s_imdsEndpoint);
```

Comment:
I think we'll end up calling ImdsEndpointUnavailable twice, once in `ImdsAvailable` and second time here.

Reply:
Thanks I meant to remove this.